closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -423,9 +423,19 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_GCMEncrypt
         OpenSSL_add_all_algorithms();
         ERR_load_crypto_strings();
         first_time_gcm = 1;
-
-        evp_gcm_cipher = EVP_aes_128_gcm();
-    } 
+    }
+    
+    switch(keyLen) {
```

Comment:
It seems the switch should assert in the default case where the keyLen is not 16, 24, 32. I'm not sure what happens when NULL is passed to EVP_CipherInit_ex(). Although maybe it sets an easily debuggable error code.